vireoSNP/vireo.py

```diff
@@ -120,67 +230,90 @@ def main():
+        cell_dat["samples"] = cell_dat["samples"][cellRange[0] : cellRange[1]]
 
     ## input donor genotype
     n_donor = options.n_donor
+
+    print((options.donor_file is not None) & (n_donor is not None))
     if options.donor_file is not None:
         if "variants" not in cell_dat.keys():
             print("No variants information is loaded, please provide base.vcf.gz")
             sys.exit(1)
 
         print("[vireo] Loading donor VCF file ...")
-        donor_vcf = load_VCF(options.donor_file, biallelic_only=True,
-                             sparse=False, format_list=[options.geno_tag])
-        
-        if (donor_vcf['n_SNP_tagged'][0] < 
-            (0.1 * len(donor_vcf['GenoINFO'][options.geno_tag]))):
-            print("[vireo] No " + options.geno_tag + " tag in donor genotype; "
-                "please try another tag for genotype, e.g., GT")
-            print("        %s" %options.donor_file)
+        donor_vcf = load_VCF(
+            options.donor_file,
+            biallelic_only=True,
+            sparse=False,
+            format_list=[options.geno_tag],
+        )
+
+        if donor_vcf["n_SNP_tagged"][0] < (
+            0.1 * len(donor_vcf["GenoINFO"][options.geno_tag])
+        ):
+            print(
+                "[vireo] No " + options.geno_tag + " tag in donor genotype; "
+                "please try another tag for genotype, e.g., GT"
+            )
+            print("        %s" % options.donor_file)
             sys.exit(1)
 
         cell_dat, donor_vcf = match_donor_VCF(cell_dat, donor_vcf)
-        donor_GPb = parse_donor_GPb(donor_vcf['GenoINFO'][options.geno_tag],
-            options.geno_tag)
-
+        donor_GPb = parse_donor_GPb(
+            donor_vcf["GenoINFO"][options.geno_tag], options.geno_tag
+        )
+        if donor_GPb is None:
+            learn_GT = False
+            donor_GPb = None
+            donor_names = ["donor%d" % x for x in range(n_donor)]
         if n_donor is None or n_donor == donor_GPb.shape[1]:
             n_donor = donor_GPb.shape[1]
-            donor_names = donor_vcf['samples']
+            donor_names = donor_vcf["samples"]
             learn_GT = False
         elif n_donor < donor_GPb.shape[1]:
             learn_GT = False
-            donor_names = ['donor%d' %x for x in range(n_donor)]
+            donor_names = ["donor%d" % x for x in range(n_donor)]
         else:
             learn_GT = True
-            donor_names = (donor_vcf['samples'] +
-                ['donor%d' %x for x in range(donor_GPb.shape[1], n_donor)])
+            donor_names = donor_vcf["samples"] + [
+                "donor%d" % x for x in range(donor_GPb.shape[1], n_donor)
+            ]
     else:
         learn_GT = True
         donor_GPb = None
-        donor_names = ['donor%d' %x for x in range(n_donor)]
+        donor_names = ["donor%d" % x for x in range(n_donor)]
+
+    if len(cell_dat["variants"]) == 0:
```

Comment:
this code checks the number of variants and mocks output if there are no variants